### PR TITLE
Add react matomo components

### LINKED ISSUE #2352: Add piwik-react-router to TM4 for Matomo Tracking
While most of our static sites are able to use the generic tracking script for integrating Matomo, TM4's react frontend requires a more specific solution. I have already found that solution in [OAM-Browser](https://github.com/hotosm/oam-browser/pull/298) and need to port that here. 

Components: 
1. Adding matomo script directly in the index.html
2. Adding a banner component and styles for the opt-out agreement
3. Adding a new variable for the matomo site_id

### PR DESCRIPTION
**Pull request by dakotabenjamin**
This is my first time writing a react component- so any suggestions to improve or do things differently greatly appreciated. 

Resolves #2352 

Based on the generic script used on our other static sites, this PR adds the matomo tracking script with two variables (already present in the deployment from TM3) and a banner component which helps the user opt-in or out of tracking if they wish. 

image

### CODE REVIEW

**Comment by willemarcel** [comment]
@dakotabenjamin I pushed some improvements to the style and to the text, replacing the styles by the css classes provided by our css framework and enabling translations.

Are you sure it's working? Do we need to add something to the App.js Router?

Probably we don't need to add the history dependency as on TM4 we use another router that already includes history: https://reach.tech/router/api/createHistory

**Comment by willemarcel** [comment]
@dakotabenjamin I also added a condition to only show the banner if there is a MATOMO_ID env var set, and modified the Privacy Policy link to use the url we have configured by ENV var. This will make it easier to other organisations to customize the use of Matomo.

**Comment by xamanu** [comment]
@dakotabenjamin please rebase.

**Comment by dakotabenjamin** [comment]
@willemarcel @xamanu finally got this rebased and working- I was missing some components. I need to update some environment variables in CircleCI before we merge so let's coordinate after :+1: review.

File: frontend/package.json [hunk 1]
```diff
@@ -36,7 +36,6 @@
     "marked": "^0.8.0",
     "node-sass": "^4.13.0",
     "osmtogeojson": "^3.0.0-beta.3",
-    "react": "^16.13.1",
```
Comment (willemarcel) [inline frontend/package.json]: I think it's not good to remove react here. We need to control its version.

Comment (dakotabenjamin) [reply frontend/package.json:39]: sorry I don't think this was intended. Will fix

**Comment by dakotabenjamin** [comment]
@willemarcel I didn't find any issues with the CSS,but also I did not notice the tachyons classes. Do you want me to reconfigure?

**Comment by willemarcel** [comment]
> @willemarcel I didn't find any issues with the CSS,but also I did not notice the tachyons classes. Do you want me to reconfigure?

@dakotabenjamin yes, I would prefer to use the tachyons CSS for it

**Comment by dakotabenjamin** [comment]
@willemarcel done- let's not forget to squash and merge

**Pull request merged**